### ISSUE TITLE
Search request + performRequestWithCompletion SIGABRT crash

### ISSUE THREAD

**Issue by nazywamsiepawel** [box/box-ios-sdk#104]
Hey,

I'm trying to implement search functionality using Box SDK. Using Swift 1.2 with Xcode 6.4.

``` swift
 func search(keyword:String) {
        var searchRequest = client.searchRequestWithQuery(keyword, inRange: NSRange(location: 0, length: 1000))

        searchRequest.performRequestWithCompletion { (items:[AnyObject]!, totalCount:UInt, range:NSRange, error:NSError!) -> Void in
          // handle processing result here.
        }
}
```

Unfortunately I'm getting a SIGABRT in `BOXAPIOperation.globalAPIOperationNetworkThreadEntryPoint` (screenshot below)

![screen shot 2015-09-14 at 2 32 04 pm](https://cloud.githubusercontent.com/assets/334875/9850894/b3420396-5aed-11e5-8686-cab726c3438a.png)

With the following error :

```
015-09-14 14:18:47.782 BoxTestApplication[52331:56352142] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil'
*** First throw call stack:
(
    0   CoreFoundation                      0x000000010c54cc65 __exceptionPreprocess + 165
    1   libobjc.A.dylib                     0x000000010b8d0bb7 objc_exception_throw + 45
    2   CoreFoundation                      0x000000010c4198ca -[__NSArrayM insertObject:atIndex:] + 954
    3   BoxTestApplication                      0x0000000107187391 __49-[BOXSearchRequest performRequestWithCompletion:]_block_invoke + 961
    4   BoxTestApplication                      0x000000010710df8a -[BOXAPIJSONOperation performCompletionCallback] + 314
    5   BoxTestApplication                      0x00000001071164ef -[BOXAPIOperation finish] + 143
    6   BoxTestApplication                      0x0000000107117080 -[BOXAPIOperation connectionDidFinishLoading:] + 128
    7   CFNetwork                           0x000000010c153d6c __65-[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:]_block_invoke + 69
    8   CFNetwork                           0x000000010c153d10 -[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:] + 199
    9   CFNetwork                           0x000000010c153e77 -[NSURLConnectionInternal _withActiveConnectionAndDelegate:] + 48
    10  CFNetwork                           0x000000010c023347 ___ZN27URLConnectionClient_Classic26_delegate_didFinishLoadingEU13block_pointerFvvE_block_invoke + 107
    11  CFNetwork                           0x000000010c0f0a71 ___ZN27URLConnectionClient_Classic18_withDelegateAsyncEPKcU13block_pointerFvP16_CFURLConnectionPK33CFURLConnectionClientCurrent_VMaxE_block_invoke_2 + 273
    12  CFNetwork                           0x000000010c00e5d6 _ZN19RunloopBlockContext13_invoke_blockEPKvPv + 72
    13  CoreFoundation                      0x000000010c453354 CFArrayApplyFunction + 68
    14  CFNetwork                           0x000000010c00e497 _ZN19RunloopBlockContext7performEv + 133
    15  CFNetwork                           0x000000010c00e2d6 _ZN17MultiplexerSource7performEv + 256
    16  CFNetwork                           0x000000010c00e0ec _ZN17MultiplexerSource8_performEPv + 72
    17  CoreFoundation                      0x000000010c480431 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
    18  CoreFoundation                      0x000000010c4762fd __CFRunLoopDoSources0 + 269
    19  CoreFoundation                      0x000000010c475934 __CFRunLoopRun + 868
    20  CoreFoundation                      0x000000010c475366 CFRunLoopRunSpecific + 470
    21  Foundation                          0x0000000108eeef92 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 275
    22  Foundation                          0x0000000108f7a080 -[NSRunLoop(NSRunLoop) run] + 74
    23  BoxTestApplication                      0x0000000107115939 +[BOXAPIOperation globalAPIOperationNetworkThreadEntryPoint:] + 105
    24  Foundation                          0x0000000108eed09d __NSThread__main__ + 1194
    25  libsystem_pthread.dylib             0x000000010d60e268 _pthread_body + 131
    26  libsystem_pthread.dylib             0x000000010d60e1e5 _pthread_body + 0
    27  libsystem_pthread.dylib             0x000000010d60c41d thread_start + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Is it a BoxSDK issue or am I doing something incorrectly?

**Comment by RicoYao** [comment]
Thank you for reporting this.

Looks like we have a bad commit from a little while back that is causing this. We're going to revert commit for now.

Pull request for reverting:
https://github.com/box/box-ios-sdk/pull/107

Offending commit:
https://github.com/box/box-ios-sdk/commit/89849ae50129acb8f8a7c16caefa6976e0e6cac2

**Comment by lasvegasgamer** [comment]
Can you please provide a status update for when the Revert PR will be merged into master?

@RicoYao

**Comment by nazywamsiepawel** [comment]
@lasvegasgamer  :+1:

**Comment by RicoYao** [comment]
Apologies for the delays guys. We'll get this merged in the next day.

**Comment by boxtcarpel** [comment]
This is now in master. A version bump release is coming soon.

**Comment by vivekvasani** [comment]
+1 for this issue. Will the update be reflected on CocoaPods?

**Comment by h2kuo** [comment]
We released v1.0.10 today. Thanks for your patience!
